lib/agent/positionsCreateWithETH.ts

```diff
@@ -0,0 +1,37 @@
+import { FallbackableOverrides } from '../common/utils/execute'
+import { Provider } from '@ethersproject/abstract-provider'
+import { swapClients } from './common/clients/swapClients'
+import {
+	approveIfNeeded as _approveIfNeeded,
+	ApproveIfNeededResult,
+} from './common/approveIfNeeded'
+import { UndefinedOr } from '@devprotocol/util-ts'
+
+type PositionsCreateWithETH = (options: {
+	readonly provider: Provider
+	readonly from: string
+	readonly propertyAddress: string
+	readonly fallbackOverrides: FallbackableOverrides
+}) => Promise<UndefinedOr<ApproveIfNeededResult>>
+
+export const positionsCreateWithETH: PositionsCreateWithETH = async (options) => {
+	const l2 = await swapClients(options.provider)
+
+	return l2 && options.fallbackOverrides.overrides?.value
+		? {
+			approveIfNeeded: _approveIfNeeded({
```

Comment:
As approve is not required for ETH transfers, this `approveIfNeeded` is not necessary.